### PR TITLE
Update WCSLIB to version 7.1

### PR DESCRIPTION
**Pull request by mcara**
This PR updates WCSLIB to version 7.1. For a complete list of changes, see https://www.atnf.csiro.au/people/mcalabre/WCS/CHANGES. The hope is that the bug fixes in this version may eliminate segmentation fault in https://github.com/astropy/astropy/pull/9641 tests.

CC: @MSeifert04 @nden

### CODE REVIEW

**Comment by bsipocz** [comment]
Given that the switch between 6.x and 7.x means ABI change, I suppose this won't be backported to LTS.  @nden - please change the milestone if you would rather prefer to it to be backported (we have precedent for wcslib updates in bugfix versions).

**Comment by Cadair** [comment]
fwiw, I am :+1: on backporting this as the solar changes would be very useful to SunPy, but I don't want to cause anyone large amounts of work!

**Comment by nden** [comment]
@bsipocz Since this has some major bug fixes I set the milestone to 4.0.1. In terms of backporting it's probably not horrible. I am willing to do it if necessary (manually, I'm not sure if you are using a script). Ping me if you want me to try this.

File: astropy/wcs/tests/test_wcs.py [hunk 1]
```diff
@@ -179,24 +198,34 @@ def test_dict_init():
     """
 
     # Dictionary with no actual WCS, returns identity transform
-    w = wcs.WCS({})
+    # with pytest.warns(FITSFixedWarning, match=r"'datfix' made the change "
+                      # r"'Set DATE-REF to '1858-11-17' from MJD-REF'."):
```
[the remark this answers is not in the document]
Comment (mcara) [reply astropy/wcs/tests/test_wcs.py]: @nden Thanks

File: astropy/wcs/wcs.py [hunk 2]
```diff
@@ -501,7 +501,12 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
         WCSBase.__init__(self, sip, cpdis, wcsprm, det2im)
 
         if fix:
-            self.fix(translate_units=translate_units)
+            if header is None:
+                with warnings.catch_warnings():
+                    warnings.simplefilter('ignore', FITSFixedWarning)
+                    self.fix(translate_units=translate_units)
```
[the remark this answers is not in the document]
Comment (mcara) [reply astropy/wcs/wcs.py]: @nden I do not think so. This `with warnings...` block applies only to the case when creating a default WCS object in particular when `header=None`. This is to remove the new warnings about `DATE-REF` when creating `WCS()`. In my opinion, creating an object with default parameters should not produce warnings.

However, I do not catch the warnings (just as before) when a header is provided, even for calls  such as `WCS({})`. So, in most other cases, all warnings will still get issued as before.

[the remark this answers is not in the document]
Comment (mcara) [reply astropy/wcs/wcs.py]: @nden I do not think so. This `with warnings...` block applies only to the case when creating a default WCS object in particular when `header=None`. This is to remove the new warnings about `DATE-REF` when creating `WCS()`. In my opinion, creating an object with default parameters should not produce warnings.

However, I do not catch the warnings (just as before) when a header is provided, even for calls  such as `WCS({})`. So, in most other cases, all warnings will still get issued as before.

[the remark this answers is not in the document]
Comment (mcara) [reply astropy/wcs/wcs.py]: @nden Or, did I miss something here?

[the remark this answers is not in the document]
Comment (nden) [reply astropy/wcs/wcs.py]: @mcara No, you are right. I was thinking about a different warning - different use case, and it's not relevant.

**Comment by bsipocz** [comment]
Backporting is not a hassle at this state, so 👍 for 4.0.1

**Comment by bsipocz** [comment]
removing changelog as #10021 makes it obsolete.

**Pull request merged**